ophyd/v2/core.py

```diff
@@ -75,6 +77,10 @@ def _run_callbacks(self, task: asyncio.Task):
             for callback in self._callbacks:
                 callback(self)
 
+    @property
+    def exception(self) -> Optional[Union[Exception, asyncio.CancelledError]]:
+        return self._exception
```

Comment:
I'd prefer to get this from the task rather than storing it...
```suggestion
        try:
            self.task.result()
        except (Exception, asyncio.CancelledError) as e:
            return e
        else:
            return None
```

Reply:
Sure, this makes sense. I'll add that in now.

Reply:
I've changed the success method, so that it doesnt store the exception in ._exception either. Presumably, the self.task attribute will contain that information in self.task.exception().

Reply:
yes